common/src/java/org/apache/hadoop/hive/common/type/TimestampUtils.java

```diff
@@ -176,7 +177,7 @@ public static Timestamp stringToTimestamp(String s) {
     // Handle simpler cases directly avoiding exceptions
     if (s.length() == DATE_LENGTH) {
       // Its a date!
-      return Timestamp.ofEpochMilli(Date.valueOf(s).toEpochMilli());
+      return Timestamp.ofEpochMilli(DateParser.parseDate(s).toEpochMilli());
```

Comment:
So, this one is a little weird.  Previously, `Date.valueOf` could potentially throw an `IllegalArgumentException` if the string was invalid, contrary to the comment:

> Handle simpler cases directly avoiding exceptions

Now, it could throw a NPE as DateParser returns `null` on invalid input.

Reply:
OK, based on a failed unit test, it does actually throw an `IllegalArugmentException` here despite the comment